appinventor/components/src/com/google/appinventor/components/runtime/AccelerometerSensor.java

```diff
@@ -485,4 +494,41 @@ public void onDelete() {
       stopListening();
     }
   }
+
+  @Override
+  public void addDataObserver(ChartDataBase dataComponent) {
+    dataSourceObservers.add(dataComponent);
+  }
+
+  @Override
+  public void removeDataObserver(ChartDataBase dataComponent) {
+    dataSourceObservers.remove(dataComponent);
+  }
+
+  @Override
+  public void notifyDataObservers(String key, Object value) {
+    // Notify each Chart Data observer component of the Data value change
+    for (ChartDataBase dataComponent : dataSourceObservers) {
+      dataComponent.onReceiveValue(this, key, value);
+    }
+  }
+
+  @Override
+  public Float getDataValue(String key) {
+    // TODO: Add documentation for X, Y and Z Data Source value names
+
+    switch (key) {
+      case "X":
```

Comment:
@ewpatton I will have to test and see, but I think you are right; This could indeed cause issues for Java versions under version 7.

But now that I think of it, maybe it would make sense to refactor the code a bit in the sensors to use maps rather than plain variables? I figure it would reduce redundancy and nested if checks in the long run.

Reply:
@ewpatton I've tested this with Android API 14 (the lowest API level supported by MPAndroidChart), seems like there were no issues there (also the API version seems to be released before Java 8). Do you think we should take extra action here with regards to even older API versions?

Reply:
@ewpatton Older API versions (namely 8) will have a different issue; Seemingly because of API support, there will be errors with regards to the MPAndroidChart library. Here's a stack trace example:
`E/AndroidRuntime(  247): java.lang.NoClassDefFoundError: com.github.mikephil.charting.charts.Chart$1
E/AndroidRuntime(  247):        at com.github.mikephil.charting.charts.Chart.init(Chart.java:211)
E/AndroidRuntime(  247):        at com.github.mikephil.charting.charts.BarLineChartBase.init(BarLineChartBase.java:153)
E/AndroidRuntime(  247):        at com.github.mikephil.charting.charts.LineChart.init(LineChart.java:32)
E/AndroidRuntime(  247):        at com.github.mikephil.charting.charts.Chart.<init>(Chart.java:184)
E/AndroidRuntime(  247):        at com.github.mikephil.charting.charts.BarLineChartBase.<init>(BarLineChartBase.java:148)
E/AndroidRuntime(  247):        at com.github.mikephil.charting.charts.LineChart.<init>(LineChart.java:19)
E/AndroidRuntime(  247):        at com.google.appinventor.components.runtime.LineChartViewBase.<init>(LineChartViewBase.java:27)
E/AndroidRuntime(  247):        at com.google.appinventor.components.runtime.LineChartView.<init>(LineChartView.java:20)
E/AndroidRuntime(  247):        at com.google.appinventor.components.runtime.Chart.createChartViewFromType(Chart.java:192)
E/AndroidRuntime(  247):        at com.google.appinventor.components.runtime.Chart.Type(Chart.java:165)
E/AndroidRuntime(  247):        at com.google.appinventor.components.runtime.Chart.<init>(Chart.java:76)
E/AndroidRuntime(  247):        at java.lang.reflect.Constructor.constructNative(Native Method)
E/AndroidRuntime(  247):        at java.lang.reflect.Constructor.newInstance(Constructor.java:446)
E/AndroidRuntime(  247):        at gnu.expr.PrimProcedure.apply(PrimProcedure.java:272)
E/AndroidRuntime(  247):        at gnu.mapping.CallContext.runUntilDone(CallContext.java:234)
E/AndroidRuntime(  247):        at gnu.mapping.CallContext.runUntilValue(CallContext.java:298)
E/AndroidRuntime(  247):        at gnu.kawa.reflect.Invoke.applyN(Invoke.java:209)
E/AndroidRuntime(  247):        at gnu.mapping.ProcedureN.apply2(ProcedureN.java:39)
E/AndroidRuntime(  247):        at appinventor.ai_test.testTinyDB.Screen1.$define(Screen1.yail:10253)
E/AndroidRuntime(  247):        at com.google.appinventor.components.runtime.Form.onCreateFinish2(Form.java:409)
E/AndroidRuntime(  247):        at com.google.appinventor.components.runtime.Form.onCreateFinish(Form.java:386)
E/AndroidRuntime(  247):        at com.google.appinventor.components.runtime.Form.onCreate(Form.java:318)
E/AndroidRuntime(  247):        at appinventor.ai_test.testTinyDB.Screen1.onCreate(Screen1.yail:10003)
E/AndroidRuntime(  247):        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1047)
E/AndroidRuntime(  247):        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2627)
E/AndroidRuntime(  247):        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2679)
E/AndroidRuntime(  247):        at android.app.ActivityThread.access$2300(ActivityThread.java:125)
E/AndroidRuntime(  247):        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2033)
E/AndroidRuntime(  247):        at android.os.Handler.dispatchMessage(Handler.java:99)
E/AndroidRuntime(  247):        at android.os.Looper.loop(Looper.java:123)
E/AndroidRuntime(  247):        at android.app.ActivityThread.main(ActivityThread.java:4627)
E/AndroidRuntime(  247):        at java.lang.reflect.Method.invokeNative(Native Method)
E/AndroidRuntime(  247):        at java.lang.reflect.Method.invoke(Method.java:521)
E/AndroidRuntime(  247):        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:868)
E/AndroidRuntime(  247):        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:626)
E/AndroidRuntime(  247):        at dalvik.system.NativeStart.main(Native Method)`